src/themes/types.ts

```diff
@@ -71,6 +71,16 @@ type EmphasisColorsStrict = Partial<{
 
 export type EmphasisColors = Extendable<EmphasisColorsStrict, ColorVariants>
 
+/**
+ * A type for extracting the color names.
+ */
+type ColorNames = keyof (EmphasisColorsStrict & NaturalColorsStrict)
+
+/**
+ * A type for extracting the color names and values.
```

Comment:
We should add a more correct description there. `Record`'s description:

> Construct a type with a set of properties K of type T

Reply:
changed